applications/ShapeOptimizationApplication/custom_utilities/response_functions/face_angle_response_function_utility.cpp

```diff
@@ -70,7 +93,9 @@ void FaceAngleResponseFunctionUtility::CalculateGradient()
 	VariableUtils().SetHistoricalVariableToZero(SHAPE_SENSITIVITY, mrModelPart.Nodes());
 
 	for (auto& cond_i : mrModelPart.Conditions()){
-
+		if (mConsiderOnlyInitiallyFeasible && !(cond_i.GetValue(FACE_ANGLE_CONSIDER))){
```

Comment:
Not for this PR. Why is this not OMP parallelized?

Reply:
Because the omp lock logic for perturbing nodes of geometries is a complex, since i would have to lock all nodes of the condition.

Can be done but is not yet until now